experiments/conductor/cmd/runner/utilities.go

```diff
@@ -134,6 +134,30 @@ func cdRepoBranchDirBash(opts *RunnerOptions, subdir string, stdin io.WriteClose
 	return msg
 }
 
+func checkoutBranchWithoutuncommittedChanges(ctx context.Context, branch Branch, workDir string) {
+	log.Print("Checking uncommitted changes: git status --porcelain")
+	statusCmd := exec.CommandContext(ctx, "git", "status", "--porcelain")
+	statusCmd.Dir = workDir
+
+	results, err := execCommand(statusCmd)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if results.Stdout != "" {
+		currentBranchCmd := exec.CommandContext(ctx, "git", "rev-parse", "--abbrev-ref", "HEAD")
+		currentBranchCmd.Dir = workDir
+
+		currentBranchResult, err := execCommand(currentBranchCmd)
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Fatalf("Found uncommitted changes at branch %q before checking out to branch %q:\n%s\n", strings.TrimSuffix(currentBranchResult.Stdout, "\n"), branch.Local, results.Stdout)
```

Comment:
This seems ok in the short term. I wonder if we could do better. 
```
$ git status --porcelain
?? .golangci.bck.yaml
```
we could delete anything on a line with `??`. Then reissue the porcelain status. If we still find a file then fatal. Later as we find more tags we improve the clean up code.

Reply:
Agreed there are better ways to handle uncommitted changes! My plan is to commit all the changes to the current branch when we see non empty stdout here. In my experience, there are two reasons I run into uncommitted changes at this point:
1. I forgot I made some changes to the WIP branch.
2. Command errored out half way and it didn't commit whatever half way done changes.

In both situations, there could be untracked and tracked files that I want to keep instead of deleting. So I'd prefer we just commit them as a `[Warning] Unfinished changes detected by conductor` commit, and then move on handling the next branch. What do you think?

Reply:
Conductor is meant to be a batch processing tool which can handle something going wrong and move on to the next task.
As such Fatal is not a good option, however having left over files which pollute the next branch are also not ok. 
So the normal behavior should be to try and clean up any left over files (logging that you did so is advisable).
If for you use case you want to stop or commit unexpected files that's fine for your use case.
However the existing steps should not be including unexpected files.

Reply:
Makes sense. I added the option for the checker to work differently based on the parameter at command level.